modules/extensions-support/src/test/java/org/mule/module/extension/internal/runtime/connector/ConnectionInterceptorTestCase.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.connector;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Answers.RETURNS_DEEP_STUBS;
+import static org.mockito.Matchers.same;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mule.module.extension.internal.ExtensionProperties.CONNECTION_PARAM;
+import org.mule.extension.runtime.OperationContext;
+import org.mule.module.extension.internal.runtime.connector.petstore.PetStoreClient;
+import org.mule.module.extension.internal.runtime.connector.petstore.PetStoreClientConnectionHandler;
+import org.mule.module.extension.internal.runtime.connector.petstore.PetStoreConnectorConfig;
+import org.mule.tck.junit4.AbstractMuleContextTestCase;
+import org.mule.tck.size.SmallTest;
+import org.mule.util.concurrent.Latch;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@SmallTest
```

Comment:
at one point I heard that it was a smalltest if it lasted less than a second which it does. But I agree with your concerns so removed.